accepted/2022/version-selection.md

```diff
@@ -0,0 +1,449 @@
+
+We decided to NOT include .NET 6 in the GitHub Ubuntu images to see if we could encourage users to migrate to `setup-dotnet`. That fixes everything, so perfect. Job done. Our users told us: ["Really?"](https://github.com/actions/virtual-environments/issues/4424#issuecomment-1012698407) They were right. We are now offering the .NET 6 SDK in Ubuntu images, as we should have from the start.
+
+Clearly, the dozen+ SDKs we're shipping isn't the right solution. However, it is also clear that relying solely on `setup-dotnet` is also not the solution. Another option is to tell everyone to use `global.json` "properly", which would mean ensuring that the existing `rollForward` property is always properly set. That's a nice idea in theory, but its really hard to require changes distributed across thousands of code-bases and a much larger set of developers. We could consider changing the default roll-forward setting.
+
+This situation is similar with Azure DevOps. We expect that other cloud and CI providers have similar challenges due to the default behavior of the .NET SDK. We're sorry about that.
+
+SDK version selection is entirely focused on repo-resident `global.json` files and that there are not adequate other controls. For example, runtime selection can be controlled by environment variables (ENVs). We could expand SDK selection to have the same broad set of controls as runtime selection. That would enable a CI build (possibly via a GitHub Actions setting) to override `global.json` in a repo, as either a temporary or permanent approach.
+
+## Migrate old apps to newer runtimes
+
+Note: This customer scenario is speculative and not based on customer feedback, but on expected future need. We'll likely choose to not productize at this time.
+
+As a development platform becomes mature and established, it leads to organizations with large deployments of apps. Certainly, we see that with .NET Framework today, and we're headed that way with .NET (Core). A runtime-based platform like .NET inherently has the challenge with compiled applications getting left behind, requiring organizations to do something to adapt those applications. Old .NET applications have hard-bound references to old runtime versions and those older applications don't want to use newer runtime versions without some encouragement. We expose good options for migrating applications (as source or binaries) to newer versions, however, they are not always convenient to use across a large body of apps. For example, applying a single version or a single roll-forward policy to thousands of apps isn't going to be an attractive choice for large organizations.
+
+Ideally, organizations could adopt the following approach for runtime management in their environments:
+
+- Adopt and deploy new .NET versions into their environments quickly, with no impact on existing applications (already the case).
```

Comment:
Did you want to cover the case of customers who want multiple apps running on multiple different runtime versions on the same server like some internal partners we have?

Reply:
Isn't that the same scenario? The implication here is that the new .NET version is for another app.